internal/provider/aws/provider.go

```diff
@@ -439,28 +440,43 @@ func (p *awsProvider) ReconcileFirewallRules(
 	log logr.Logger,
 	nodeName, instanceID string,
 	firewallRules []v1alpha1.FirewallRule,
-) (v1alpha1.FirewallRuleStatus, error) {
+) (v1alpha1.FirewallRuleStatus, []kmetav1.Condition, error) {
 	status := v1alpha1.FirewallRuleStatus{
-		State: v1alpha1.FirewallRuleStateNone,
+		State: v1alpha1.FirewallRuleStatePending,
 	}
 
+	conditions := []kmetav1.Condition{}
+
 	// Get the instance
 	instance, err := p.getInstance(ctx, instanceID)
 	if err != nil {
-		return status, fmt.Errorf("failed to get instance: %w", err)
+		return status, conditions, fmt.Errorf("failed to get instance: %w", err)
 	}
 
 	// Get the security group associated with the instance
 	securityGroup, err := p.getSecurityGroup(ctx, Managed(), WithVPCID(aws.StringValue(instance.VpcId)), WithNodeName(nodeName))
 	if err != nil && err.(*provider.Error).Code != provider.NotFoundError {
-		return status, fmt.Errorf("failed to get security group: %w", err)
+		conditions = append(conditions, kmetav1.Condition{
+			Type:    v1alpha1.FirewallRuleConditionTypeSecurityGroupCreated,
+			Status:  kmetav1.ConditionFalse,
+			Reason:  v1alpha1.FirewallRuleConditionReasonSecurityGroupNotFound,
+			Message: "Security group not found",
+		})
+		return status, conditions, fmt.Errorf("failed to get security group: %w", err)
```

Comment:
Here you're in the case `Code != provider.NotFoundError`, so you 
```suggestion
		conditions = append(conditions, kmetav1.Condition{
			Type:    v1alpha1.FirewallRuleConditionTypeSecurityGroupCreated,
			Status:  kmetav1.ConditionUnknown,
			Reason:  v1alpha1.FirewallRuleConditionReasonSecurityGroupNotFound,
			Message: "Fail to retrieve security group from AWS",
		})
		return status, conditions, fmt.Errorf("failed to get security group: %w", err)
```